### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by Recillah-Khamala**
### **Blog app - controllers specs**

Hello code reviewer

In this milestone, I have-:

- Created spec files for all controllers.
- Made sure that for each action I checked:
  - If the response status is correct.
  - If a correct template is rendered.
  - If the response body includes the correct placeholder text.

### CODE REVIEW

File: spec/support/factory_bot.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# # spec/support/factory_bot.rb
+# spec/support/factory_bot.rb
 
-# RSpec.configure do |config|
-#   config.include FactoryBot::Syntax::Methods
-# end
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
```
Comment (FrederickMih) [inline spec/support/factory_bot.rb:5]: Capture


- [x] I'm unable to set up your app locally through `rails db:setup` as a result of the error above. Kindly add the line ` require 'rspec/rails'`  or `require 'rspec/core'` at the top of spec/support/factory_bot.rb  file

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi 🙋 @Recillah-Khamala , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights 🟢 

- PR is professional ✔️ 
- Descripitive README.md file ✔️ 
- All Linter checks passed ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    it 'renders the index template' do
+      get :index, params: { user_id: user.id }
+      expect(response).to render_template(:index)
+    end
+
+    it 'assigns the correct post count' do
+      get :index, params: { user_id: user.id }
+      expect(assigns(:posts).count).to be > 1
+    end
+  end
```
Comment (TedLivist) [inline spec/controllers/posts_controller_spec.rb]: - [x] This test is failing. Kindly fix that.

  image

File: spec/support/factory_bot.rb (hunk 1 above)
Comment (Recillah-Khamala) [reply spec/support/factory_bot.rb:5]: Change done as per your instruction

File: spec/controllers/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
```
Comment (TedLivist) [inline spec/controllers/posts_controller_spec.rb]: - [ ] **[OPTIONAL]** Tests of the `type: :controller` are considered obsolete in RSpec team. You should consider using `type: :request` instead.

  [Link to notice](https://stackoverflow.com/a/46500842)

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
```
Comment (TedLivist) [inline spec/controllers/users_controller_spec.rb]: - [ ] **[OPTIONAL]** Tests of the `type: :controller` are considered obsolete in RSpec team. You should consider using `type: :request` instead.

  [Link to notice](https://stackoverflow.com/a/46500842)

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Recillah-Khamala,

✔️ Some passing tests

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Recillah-Khamala 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights 💯**
- Well done on making the required changes from the previous reviewers
- Extras on making the "Optional" changes from the previous reviewer 🥇 

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/posts_controller_spec.rb (hunk 2 above)
Comment (Recillah-Khamala) [reply spec/controllers/posts_controller_spec.rb]: Test now passing

File: spec/controllers/users_controller_spec.rb (hunk 4 above)
Comment (Recillah-Khamala) [reply spec/controllers/users_controller_spec.rb]: moved to the resource folder

File: spec/controllers/posts_controller_spec.rb (hunk 3 above)
Comment (Recillah-Khamala) [reply spec/controllers/posts_controller_spec.rb]: moved to the resource folder

**Pull request merged**